moz_preprocess/weekly_aggregate.sql

```diff
@@ -0,0 +1,472 @@
+    SUM(
+    IF
+      (REGEXP_CONTAINS(LOWER(engine), "duckduckgo|ddg"), COALESCE(ad_click_organic, 0), 0)) AS ddg_ad_click_organic,
+    SUM(
+    IF
+      (REGEXP_CONTAINS(LOWER(engine), "duckduckgo|ddg"), COALESCE(search_with_ads, 0), 0)) AS ddg_search_with_ads,
+    SUM(
+    IF
+      (REGEXP_CONTAINS(LOWER(engine), "duckduckgo|ddg"), COALESCE(search_with_ads_organic, 0), 0)) AS ddg_search_with_ads_organic,
+    /* These metrics are not yet implemented.
+    SUM(
+    IF
+      (LOWER(engine) LIKE "%google%"
+        AND is_sap_monetizable
+        AND NOT has_adblocker_addon, COALESCE(tagged_sap, 0), 0)) AS google_monetizable_tagged_sap,
+    SUM(
+    IF
+      (LOWER(engine) LIKE "%google%"
+        AND is_sap_monetizable
+        AND NOT has_adblocker_addon, COALESCE(tagged_follow_on, 0), 0)) AS google_monetizable_tagged_follow_on,
+    SUM(
+    IF
+      (LOWER(engine) LIKE "%google%"
+        AND is_sap_monetizable
+        AND NOT has_adblocker_addon, COALESCE(sap, 0), 0)) AS google_monetizable_sap,
+    SUM(
+    IF
+      (LOWER(engine) LIKE "%bing%"
+        AND is_sap_monetizable
+        AND NOT has_adblocker_addon, COALESCE(tagged_sap, 0), 0)) AS bing_monetizable_tagged_sap,
+    SUM(
+    IF
+      (LOWER(engine) LIKE "%bing%"
+        AND is_sap_monetizable
+        AND NOT has_adblocker_addon, COALESCE(tagged_follow_on, 0), 0)) AS bing_monetizable_tagged_follow_on,
+    SUM(
+    IF
+      (LOWER(engine) LIKE "%bing%"
+        AND is_sap_monetizable
+        AND NOT has_adblocker_addon, COALESCE(sap, 0), 0)) AS bing_monetizable_sap,
+    SUM(
+    IF
+      (REGEXP_CONTAINS(LOWER(engine), "duckduckgo|ddg")
+        AND is_sap_monetizable
+        AND NOT has_adblocker_addon, COALESCE(tagged_sap, 0), 0)) AS ddg_monetizable_tagged_sap,
+    SUM(
+    IF
+      (REGEXP_CONTAINS(LOWER(engine), "duckduckgo|ddg")
+        AND is_sap_monetizable
+        AND NOT has_adblocker_addon, COALESCE(tagged_follow_on, 0), 0)) AS ddg_monetizable_tagged_follow_on,
+    SUM(
+    IF
+      (REGEXP_CONTAINS(LOWER(engine), "duckduckgo|ddg")
+        AND is_sap_monetizable
+        AND NOT has_adblocker_addon, COALESCE(sap, 0), 0)) AS ddg_monetizable_sap,
+    */
+    --Aggregate by summing.
+    SUM(COALESCE(organic, 0)) AS organic,
+    SUM(COALESCE(tagged_sap, 0)) AS tagged_sap,
+    SUM(COALESCE(tagged_follow_on, 0)) AS tagged_follow_on,
+    SUM(COALESCE(sap, 0)) AS sap,
+    SUM(COALESCE(ad_click, 0)) AS ad_click,
+    SUM(COALESCE(ad_click_organic, 0)) AS ad_click_organic,
+    SUM(COALESCE(search_with_ads, 0)) AS search_with_ads,
+    SUM(COALESCE(search_with_ads_organic, 0)) AS search_with_ads_organic,
+    /* These metrics are not yet implemented.
+    --Aggregate by taking any value. They are all the same.
+    ANY_VALUE(has_adblocker_addon) AS has_adblocker_addon,
+    */
+  FROM
+    `mozdata.search.search_clients_engines_sources_daily`
+  WHERE
+    submission_date >= '@week_start_date'
```

Comment:
This (and the other `@strings`) could be replaced by `'{week_start_date}'`

Reply:
done :)